test/e2e/scalenodes.go

```diff
@@ -75,11 +81,66 @@ var _ = Describe("Scale nodes", func() {
 		err = waitForScale(mss.Items[0].Name)
 		Expect(err).NotTo(HaveOccurred())
 	})
+
+	Specify("operator should maintain at least three worker replicas", func() {
+		ctx := context.Background()
+		instance, err := clients.AROClusters.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		mss, err := clients.MachineAPI.MachineV1beta1().MachineSets(machineSetsNamespace).List(ctx, metav1.ListOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(mss.Items).NotTo(BeEmpty())
+
+		// we don't want to run subsequent tests if there are extra custom machinesets (unlikely in E2E, but possible)
+		isCustom := false
+		for _, machineset := range mss.Items {
+			if !strings.Contains(machineset.Name, instance.Spec.InfraID) {
+				isCustom = true
+			}
+		}
+
+		switch !isCustom {
+		case len(mss.Items) == 3:
+			// cluster has AZs, remove one replica from two machinesets
+			err = scale(mss.Items[0].Name, -1)
+			Expect(err).NotTo(HaveOccurred())
+
+			err = waitForScale(mss.Items[0].Name)
+			Expect(err).NotTo(HaveOccurred())
+
+			err = scale(mss.Items[1].Name, -1)
+			Expect(err).NotTo(HaveOccurred())
+
+			err = waitForScale(mss.Items[1].Name)
+			Expect(err).NotTo(HaveOccurred())
+
+			// expect the operator to have added a replica back
+			ms, err := clients.MachineAPI.MachineV1beta1().MachineSets(machineSetsNamespace).Get(ctx, mss.Items[1].Name, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(ms.Spec.Replicas).To(BeEquivalentTo(1))
+
+			// revert the cluster to previous state
+			err = scale(mss.Items[0].Name, 1)
+			Expect(err).NotTo(HaveOccurred())
+
+		case len(mss.Items) < 3:
```

Comment:
This is annoying, but hopefully we will only have scaling in the cases and assertion after the `switch`